File: solutions/python/bottle-song/1/bottle_song.py
numbers = ["ten", "nine", "eight", "seven", "six", "five", "four", "three", "two", "one", "no"]

def recite(start, take=1):
    song = []
    for idx in range(10-start, 10-start+take):
        btl_ps = "bottle"
        if idx!=9:
            btl_ps +="s"
        btl_ps_next = "bottle"
        if idx!=8:
            btl_ps_next += "s"
        number1 = numbers[idx].capitalize()
        number2 = numbers[idx+1]
        song.extend(
            ["{0} green {1} hanging on the wall,".format(number1, btl_ps), "{0} green {1} hanging on the wall,".format(number1, btl_ps),
            "And if one green bottle should accidentally fall,",
            "There'll be {} green {} hanging on the wall.".format(number2, btl_ps_next)])
        if take!=1 and idx!=9-start+take:
            song.append("")
    return song
           

File: solutions/dart/bottle-song/1/lib/bottle_song.dart
class BottleSong {
  static List<String> numbers = ["ten", "nine", "eight", "seven", "six", "five", "four", "three", "two", "one", "no"];

  static  String occurence = 'And if one green bottle should accidentally fall,';


  String capitalize(String word){
    return word[0].toUpperCase() + word.substring(1);
  }

  List<String> recite(int start, int take){
    var song = <String>[];
    var first = 10-start;
    var last = 10 - start + take;
    for(var idx=first; idx<last; idx++){
      String currentBtls = 'bottle';
      currentBtls += (idx!=9) ? 's' : '';
      String nextBtls = 'bottle';
      nextBtls += (idx!=8) ? 's' : '';
      String number1 = capitalize(numbers[idx]);
      String number2 = numbers[idx+1];
      String firstLine = '${number1} green ${currentBtls} hanging on the wall,';
      song.addAll([firstLine, firstLine]);
      song.add(occurence);
      song.add("There'll be ${number2} green ${nextBtls} hanging on the wall.");        
      if(take!=1 && idx!=9-start+take)
        song.add('');  
    }
    return song;
  }  
}
